src/methods/join.js

```diff
@@ -286,67 +286,107 @@ export class JoinImplementation {
     // effects collected after pnc was constructed
     e: Effects
   ): ForkedAbruptCompletion {
+    let recurse = (x: PossiblyNormalCompletion, xe: Effects): [ForkedAbruptCompletion, Effects] => {
+      let nx = this.replacePossiblyNormalCompletionWithForkedAbruptCompletion(realm, x, ac, e);
+      let nxe = new Effects(nx, xe.generator, xe.modifiedBindings, xe.modifiedProperties, xe.createdObjects);
+      return [nx, nxe];
+    };
+
+    let cloneEffects = () => {
```

Comment:
I would prefer a function, rather than a procedure with side effects.

Reply:
I removed the side effects. Since you’ll be working on this code, I’ll let you move it to the appropriate scope 👍